pkg/serverless/trace/inferredspan/span_enrichment.go

```diff
@@ -241,6 +354,29 @@ func (inferredSpan *InferredSpan) EnrichInferredSpanWithSQSEvent(eventPayload ev
 		receiptHandle:  eventRecord.ReceiptHandle,
 		senderID:       eventRecord.Attributes["SenderId"],
 	}
+	if ddMessageAttribute, ok := eventRecord.MessageAttributes["_datadog"]; ok {
+		traceID, parentID, samplingPriority, err = extractContextFromPureSqsEvent(ddMessageAttribute)
+	} else {
+		traceID, parentID, samplingPriority, err = extractContextFromSNSSQSEvent(eventRecord)
+	}
+
+	if err != nil {
+		log.Errorf("Failed to extract context: %v", err)
+		return
+	}
+
+	uint64TraceID, uint64ParentID, samplingPriorityFloat64, err = convertValues(traceID, parentID, samplingPriority)
+	if err != nil {
```

Comment:
Typically we'd do this and line 373 in the same conditional, no need to evaluate twice